src/memo/memo.ml

```diff
@@ -726,26 +764,36 @@ end = struct
       Call_stack.push_sync_frame frame (fun () ->
           let from_cache =
             match dep_node.last_cached_value with
-            | None ->
-              Prev_cycle_cache_lookup_result.Invalid { old_value = None }
+            | None -> Prev_cycle_cache_lookup_result.Invalid No_value
             | Some cv -> (
-              let res = deps_changed cv.deps in
-              match res with
-              | Unchanged ->
-                cv.last_validated_at <- Run.current ();
-                Prev_cycle_cache_lookup_result.Valid cv
-              | Changed ->
-                dep_node.last_cached_value <- None;
-                Invalid { old_value = Some cv } )
+              match Result.is_error cv.value with
+              | true -> Invalid No_value
```

Comment:
Maybe add the following comment?

(* For errors the dependencies are not always recorded correctly. 
  In particular, dependency cycle errors have missing deps.
  Because of this, we can't use `deps_changed` in this case. *)

Reply:
Agreed with both points, done!